% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/get_gene_expression.R
\name{get_gene_expression}
\alias{get_gene_expression}
\title{Get Gene Expression}
\usage{
get_gene_expression(
  gencodeIds,
  datasetId = "gtex_v8",
  tissueSiteDetailId = NULL,
  attributeSubset = NULL,
  page = 0,
  itemsPerPage = 250
)
}
\arguments{
\item{gencodeIds}{A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").}

\item{datasetId}{String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".
Default = "gtex_v8".}

\item{tissueSiteDetailId}{String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. "Whole_Blood"; use \code{\link[=available_tissueSiteDetailIds]{available_tissueSiteDetailIds()}}
to see valid values) or an Ontology ID.}

\item{page}{Integer (default = 0).}

\item{itemsPerPage}{Integer (default = 250).}
}
\value{
A Tibble
}
\description{
\itemize{
\item Find normalized gene expression data.
\item Returns normalized gene expression in tissues at the sample level.
\item Results may be filtered by dataset, gene or tissue, but at least one gene must be provided.
\item By default, this service queries the latest GTEx release.
}

\href{https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_gene_expression_api_v2_expression_geneExpression_get}{GTEx Portal API documentation}
}
\examples{
\dontrun{
get_gene_expression(gencodeIds = "ENSG00000132693.12")
}
}
\seealso{
Other Expression Data Endpoints: 
\code{\link{get_clustered_median_exon_expression}()},
\code{\link{get_clustered_median_gene_expression}()},
\code{\link{get_clustered_median_junction_expression}()},
\code{\link{get_clustered_median_transcript_expression}()},
\code{\link{get_expression_pca}()},
\code{\link{get_median_exon_expression}()},
\code{\link{get_median_gene_expression}()},
\code{\link{get_median_junction_expression}()},
\code{\link{get_median_transcript_expression}()},
\code{\link{get_single_nucleus_gex_summary}()},
\code{\link{get_single_nucleus_gex}()},
\code{\link{get_top_expressed_genes}()}
}
\concept{Expression Data Endpoints}
