<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Calculate your own eQTLs
This service calculates the gene-variant association for any given pair of gene and variant, which may or may not be significant.
This requires as input a GENCODE ID, GTEx variant ID, and tissue site detail ID.


By default, the calculation is based on the latest GTEx release.
GTEx Portal API documentation."><title>Calculate Expression Quantitative Trait Loci — calculate_expression_quantitative_trait_loci • gtexr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Calculate Expression Quantitative Trait Loci — calculate_expression_quantitative_trait_loci"><meta property="og:description" content="Calculate your own eQTLs
This service calculates the gene-variant association for any given pair of gene and variant, which may or may not be significant.
This requires as input a GENCODE ID, GTEx variant ID, and tissue site detail ID.


By default, the calculation is based on the latest GTEx release.
GTEx Portal API documentation."><meta property="og:image" content="https://rmgpanw.github.io/gtexr/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/gtexr.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/developer_guide.html">Developer guide</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/rmgpanw/gtexr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Calculate Expression Quantitative Trait Loci</h1>
      <small class="dont-index">Source: <a href="https://github.com/rmgpanw/gtexr/blob/HEAD/R/calculate_expression_quantitative_trait_loci.R" class="external-link"><code>R/calculate_expression_quantitative_trait_loci.R</code></a></small>
      <div class="d-none name"><code>calculate_expression_quantitative_trait_loci.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Calculate your own eQTLs</p><ul><li><p>This service calculates the gene-variant association for any given pair of gene and variant, which may or may not be significant.</p></li>
<li><p>This requires as input a GENCODE ID, GTEx variant ID, and tissue site detail ID.</p></li>
</ul><p>By default, the calculation is based on the latest GTEx release.</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Dynamic-Association-Endpoints/operation/calculate_expression_quantitative_trait_loci_api_v2_association_dyneqtl_get" class="external-link">GTEx Portal API documentation</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">calculate_expression_quantitative_trait_loci</span><span class="op">(</span></span>
<span>  <span class="va">tissueSiteDetailId</span>,</span>
<span>  <span class="va">gencodeId</span>,</span>
<span>  <span class="va">variantId</span>,</span>
<span>  datasetId <span class="op">=</span> <span class="st">"gtex_v8"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>tissueSiteDetailId</dt>
<dd><p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. "Whole_Blood"; use <code><a href="available_tissueSiteDetailIds.html">available_tissueSiteDetailIds()</a></code>
to see valid values) or an Ontology ID.</p></dd>


<dt>gencodeId</dt>
<dd><p>String. A Versioned GENCODE ID of a gene, e.g.
"ENSG00000065613.9".</p></dd>


<dt>variantId</dt>
<dd><p>String. A gtex variant ID.</p></dd>


<dt>datasetId</dt>
<dd><p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A tibble.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Notes on output:</p><ul><li><p>Beta and standard error are recorded in columns <code>nes</code> and <code>error</code> respectively (see <a href="https://gtexportal.org/home/faq#nes_beta" class="external-link">GTEx FAQs</a>)</p></li>
<li><p><code>variantId</code> contains (in order) chromosome, position, reference allele, alternative allele and human genome build separated by underscores. The reference and alternative alleles for "chr1_13550_G_A_b38" for example are "G" and "A" respectively.</p></li>
<li><p>See examples for how to calculate minor and alternative allele frequencies.</p></li>
</ul><p>Notes on input:</p><ul><li><p>Argument <code>variantId</code> also accepts RSIDs.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Dynamic Association Endpoints: 
<code><a href="calculate_ieqtls.html">calculate_ieqtls</a>()</code>,
<code><a href="calculate_isqtls.html">calculate_isqtls</a>()</code>,
<code><a href="calculate_splicing_quantitative_trait_loci.html">calculate_splicing_quantitative_trait_loci</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># perform request - returns a tibble with a single row</span></span></span>
<span class="r-in"><span><span class="fu">calculate_expression_quantitative_trait_loci</span><span class="op">(</span>tissueSiteDetailId <span class="op">=</span> <span class="st">"Whole_Blood"</span>,</span></span>
<span class="r-in"><span>                                             gencodeId <span class="op">=</span> <span class="st">"ENSG00000203782.5"</span>,</span></span>
<span class="r-in"><span>                                             variantId <span class="op">=</span> <span class="st">"rs79641866"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 15</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   data               error gencodeId  geneSymbol genotypes hetCount homoAltCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;list&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> <span style="color: #949494;">&lt;tibble [670 × 1]&gt;</span> 0.148 ENSG00000… LOR        <span style="color: #949494;">&lt;tibble&gt;</span>        38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 8 more variables: homoRefCount &lt;int&gt;, maf &lt;dbl&gt;, nes &lt;dbl&gt;, pValue &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   pValueThreshold &lt;dbl&gt;, tStatistic &lt;dbl&gt;, tissueSiteDetailId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   variantId &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># unnest list columns with tidyr::unnest()</span></span></span>
<span class="r-in"><span><span class="fu">calculate_expression_quantitative_trait_loci</span><span class="op">(</span>tissueSiteDetailId <span class="op">=</span> <span class="st">"Whole_Blood"</span>,</span></span>
<span class="r-in"><span>                                             gencodeId <span class="op">=</span> <span class="st">"ENSG00000203782.5"</span>,</span></span>
<span class="r-in"><span>                                             variantId <span class="op">=</span> <span class="st">"rs79641866"</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/unnest.html" class="external-link">unnest</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"data"</span>, <span class="st">"genotypes"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 670 × 15</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        data error gencodeId         geneSymbol genotypes hetCount homoAltCount</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> -<span style="color: #BB0000;">0.118</span>   0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> -<span style="color: #BB0000;">0.321</span>   0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span>  0.266   0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span>  0.061<span style="text-decoration: underline;">7</span>  0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> -<span style="color: #BB0000;">0.009</span><span style="color: #BB0000; text-decoration: underline;">34</span> 0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span>  0.678   0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> -<span style="color: #BB0000;">0.305</span>   0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> -<span style="color: #BB0000;">0.500</span>   0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> -<span style="color: #BB0000;">0.046</span><span style="color: #BB0000; text-decoration: underline;">7</span>  0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> -<span style="color: #BB0000;">0.205</span>   0.148 ENSG00000203782.5 LOR                0       38            0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 660 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 8 more variables: homoRefCount &lt;int&gt;, maf &lt;dbl&gt;, nes &lt;dbl&gt;, pValue &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   pValueThreshold &lt;dbl&gt;, tStatistic &lt;dbl&gt;, tissueSiteDetailId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   variantId &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># to calculate minor and alternative allele frequencies</span></span></span>
<span class="r-in"><span><span class="fu">calculate_expression_quantitative_trait_loci</span><span class="op">(</span></span></span>
<span class="r-in"><span>  tissueSiteDetailId <span class="op">=</span> <span class="st">"Liver"</span>,</span></span>
<span class="r-in"><span>  gencodeId <span class="op">=</span> <span class="st">"ENSG00000237973.1"</span>,</span></span>
<span class="r-in"><span>  variantId <span class="op">=</span> <span class="st">"rs12119111"</span></span></span>
<span class="r-in"><span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span>.id <span class="op">=</span> <span class="st">"rsid"</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>   col <span class="op">=</span> <span class="st">"variantId"</span>,</span></span>
<span class="r-in"><span>   into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>     <span class="st">"chromosome"</span>,</span></span>
<span class="r-in"><span>     <span class="st">"position"</span>,</span></span>
<span class="r-in"><span>     <span class="st">"reference_allele"</span>,</span></span>
<span class="r-in"><span>     <span class="st">"alternative_allele"</span>,</span></span>
<span class="r-in"><span>     <span class="st">"genome_build"</span></span></span>
<span class="r-in"><span>   <span class="op">)</span>,</span></span>
<span class="r-in"><span>   sep <span class="op">=</span> <span class="st">"_"</span></span></span>
<span class="r-in"><span> <span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="co"># ...then ascertain alternative_allele frequency</span></span></span>
<span class="r-in"><span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>   alt_allele_count <span class="op">=</span> <span class="op">(</span><span class="fl">2</span> <span class="op">*</span> <span class="va">homoAltCount</span><span class="op">)</span> <span class="op">+</span> <span class="va">hetCount</span>,</span></span>
<span class="r-in"><span>   total_allele_count <span class="op">=</span> <span class="fl">2</span> <span class="op">*</span> <span class="op">(</span><span class="va">homoAltCount</span> <span class="op">+</span> <span class="va">hetCount</span> <span class="op">+</span>  <span class="va">homoRefCount</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>   alternative_allele_frequency <span class="op">=</span> <span class="va">alt_allele_count</span> <span class="op">/</span> <span class="va">total_allele_count</span></span></span>
<span class="r-in"><span> <span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>   <span class="va">rsid</span>,</span></span>
<span class="r-in"><span>   beta <span class="op">=</span> <span class="va">nes</span>,</span></span>
<span class="r-in"><span>   se <span class="op">=</span> <span class="va">error</span>,</span></span>
<span class="r-in"><span>   <span class="va">pValue</span>,</span></span>
<span class="r-in"><span>   minor_allele_frequency <span class="op">=</span> <span class="va">maf</span>,</span></span>
<span class="r-in"><span>   <span class="va">alternative_allele_frequency</span>,</span></span>
<span class="r-in"><span>   <span class="va">chromosome</span><span class="op">:</span><span class="va">genome_build</span>,</span></span>
<span class="r-in"><span>   <span class="va">tissueSiteDetailId</span></span></span>
<span class="r-in"><span> <span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 12</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   rsid    beta     se pValue minor_allele_frequency alternative_allele_frequency</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 1     0.027<span style="text-decoration: underline;">0</span> 0.067<span style="text-decoration: underline;">0</span>  0.688                  0.365                        0.635</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 6 more variables: chromosome &lt;chr&gt;, position &lt;chr&gt;, reference_allele &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   alternative_allele &lt;chr&gt;, genome_build &lt;chr&gt;, tissueSiteDetailId &lt;chr&gt;</span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

