<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get Significant Single Tissue Eqtls — get_significant_single_tissue_eqtls • gtexr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Significant Single Tissue Eqtls — get_significant_single_tissue_eqtls"><meta name="description" content="Find significant single tissue
eQTLs.
This service returns precomputed significant single tissue eQTLs.
Results may be filtered by tissue, gene, variant or dataset.
To search by gene, use the versioned GENCODE ID.
To search by variant, use the dbSNP rs ID (snpId).


By default, the service queries the latest GTEx release and the retrieved data is split into pages with items_per_page entries per page
GTEx Portal API documentation."><meta property="og:description" content="Find significant single tissue
eQTLs.
This service returns precomputed significant single tissue eQTLs.
Results may be filtered by tissue, gene, variant or dataset.
To search by gene, use the versioned GENCODE ID.
To search by variant, use the dbSNP rs ID (snpId).


By default, the service queries the latest GTEx release and the retrieved data is split into pages with items_per_page entries per page
GTEx Portal API documentation."><meta property="og:image" content="https://rmgpanw.github.io/gtexr/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gtexr.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/developer_guide.html">Developer guide</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/rmgpanw/gtexr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get Significant Single Tissue Eqtls</h1>
      <small class="dont-index">Source: <a href="https://github.com/rmgpanw/gtexr/blob/main/R/get_significant_single_tissue_eqtls.R" class="external-link"><code>R/get_significant_single_tissue_eqtls.R</code></a></small>
      <div class="d-none name"><code>get_significant_single_tissue_eqtls.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Find significant single tissue
eQTLs.</p><ul><li><p>This service returns precomputed significant single tissue eQTLs.</p></li>
<li><p>Results may be filtered by tissue, gene, variant or dataset.</p></li>
<li><p>To search by gene, use the versioned GENCODE ID.</p></li>
<li><p>To search by variant, use the dbSNP rs ID (snpId).</p></li>
</ul><p>By default, the service queries the latest GTEx release and the retrieved data is split into pages with <code>items_per_page</code> entries per page</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_significant_single_tissue_eqtls_api_v2_association_singleTissueEqtl_get" class="external-link">GTEx Portal API documentation</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_significant_single_tissue_eqtls</span><span class="op">(</span></span>
<span>  gencodeIds <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  variantIds <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tissueSiteDetailIds <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  datasetId <span class="op">=</span> <span class="st">"gtex_v8"</span>,</span>
<span>  page <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  itemsPerPage <span class="op">=</span> <span class="fl">250</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gencodeids">gencodeIds<a class="anchor" aria-label="anchor" href="#arg-gencodeids"></a></dt>
<dd><p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p></dd>


<dt id="arg-variantids">variantIds<a class="anchor" aria-label="anchor" href="#arg-variantids"></a></dt>
<dd><p>Character vector. Gtex variant IDs.</p></dd>


<dt id="arg-tissuesitedetailids">tissueSiteDetailIds<a class="anchor" aria-label="anchor" href="#arg-tissuesitedetailids"></a></dt>
<dd><p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code><a href="get_tissue_site_detail.html">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p></dd>


<dt id="arg-datasetid">datasetId<a class="anchor" aria-label="anchor" href="#arg-datasetid"></a></dt>
<dd><p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p></dd>


<dt id="arg-page">page<a class="anchor" aria-label="anchor" href="#arg-page"></a></dt>
<dd><p>Integer (default = 0).</p></dd>


<dt id="arg-itemsperpage">itemsPerPage<a class="anchor" aria-label="anchor" href="#arg-itemsperpage"></a></dt>
<dd><p>Integer (default = 250).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A tibble.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p><strong>Note:</strong> although the GTEx Portal API documentation says to use the dbSNP rsID
when searching by variant, this returns no results. Instead use gtex variant
IDs e.g. use "chr1_153209640_C_A_b38" instead of "rs1410858".</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Static Association Endpoints:
<code><a href="get_eqtl_genes.html">get_eqtl_genes</a>()</code>,
<code><a href="get_fine_mapping.html">get_fine_mapping</a>()</code>,
<code><a href="get_independent_eqtl.html">get_independent_eqtl</a>()</code>,
<code><a href="get_multi_tissue_eqtls.html">get_multi_tissue_eqtls</a>()</code>,
<code><a href="get_significant_single_tissue_eqtls_by_location.html">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="get_significant_single_tissue_ieqtls.html">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="get_significant_single_tissue_isqtls.html">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="get_significant_single_tissue_sqtls.html">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="get_sqtl_genes.html">get_sqtl_genes</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># search by gene</span></span></span>
<span class="r-in"><span><span class="fu">get_significant_single_tissue_eqtls</span><span class="op">(</span>gencodeIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000132693.12"</span>,</span></span>
<span class="r-in"><span>                                                  <span class="st">"ENSG00000203782.5"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> numberOfPages = 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> page = 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> totalNumberOfItems = 249</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 249 × 13</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    snpId            pos snpIdUpper variantId  geneSymbol  pValue geneSymbolUpper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> rs12128960 159<span style="text-decoration: underline;">343</span>657 RS12128960 chr1_1593… CRP        8.52<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> rs12132451 159<span style="text-decoration: underline;">344</span>052 RS12132451 chr1_1593… CRP        7.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> rs12136402 159<span style="text-decoration: underline;">347</span>493 RS12136402 chr1_1593… CRP        7.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> rs10908709 159<span style="text-decoration: underline;">350</span>390 RS10908709 chr1_1593… CRP        7.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> rs10908710 159<span style="text-decoration: underline;">351</span>189 RS10908710 chr1_1593… CRP        7.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> rs11265178 159<span style="text-decoration: underline;">359</span>256 RS11265178 chr1_1593… CRP        9.62<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> rs35532309 159<span style="text-decoration: underline;">360</span>755 RS35532309 chr1_1593… CRP        6.11<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> rs6692378  159<span style="text-decoration: underline;">369</span>451 RS6692378  chr1_1593… CRP        1.17<span style="color: #949494;">e</span><span style="color: #BB0000;">-6</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> rs10908714 159<span style="text-decoration: underline;">370</span>563 RS10908714 chr1_1593… CRP        1.80<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> rs6656924  159<span style="text-decoration: underline;">372</span>915 RS6656924  chr1_1593… CRP        1.00<span style="color: #949494;">e</span><span style="color: #BB0000;">-6</span> CRP            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 239 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 6 more variables: datasetId &lt;chr&gt;, tissueSiteDetailId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ontologyId &lt;chr&gt;, chromosome &lt;chr&gt;, gencodeId &lt;chr&gt;, nes &lt;dbl&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># search by variant - must be variantId (not rsid)</span></span></span>
<span class="r-in"><span><span class="fu">get_significant_single_tissue_eqtls</span><span class="op">(</span>variantIds <span class="op">=</span> <span class="st">"chr1_153209640_C_A_b38"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> numberOfPages = 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> page = 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> totalNumberOfItems = 2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 13</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   snpId           pos snpIdUpper variantId   geneSymbol   pValue geneSymbolUpper</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> rs1410858 153<span style="text-decoration: underline;">209</span>640 RS1410858  chr1_15320… LOR        5.47<span style="color: #949494;">e</span><span style="color: #BB0000;">- 5</span> LOR            </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> rs1410858 153<span style="text-decoration: underline;">209</span>640 RS1410858  chr1_15320… PRR9       5.35<span style="color: #949494;">e</span><span style="color: #BB0000;">-19</span> PRR9           </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 6 more variables: datasetId &lt;chr&gt;, tissueSiteDetailId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   ontologyId &lt;chr&gt;, chromosome &lt;chr&gt;, gencodeId &lt;chr&gt;, nes &lt;dbl&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># filter by gene/variant and tissue site - either `gencodeIds` or `variantIds`</span></span></span>
<span class="r-in"><span><span class="co"># should be supplied as a minimum</span></span></span>
<span class="r-in"><span><span class="fu">get_significant_single_tissue_eqtls</span><span class="op">(</span>gencodeIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000132693.12"</span>,</span></span>
<span class="r-in"><span>                                                  <span class="st">"ENSG00000203782.5"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                                    variantIds <span class="op">=</span> <span class="st">"chr1_153209640_C_A_b38"</span>,</span></span>
<span class="r-in"><span>                                    tissueSiteDetailIds <span class="op">=</span> <span class="st">"Whole_Blood"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> numberOfPages = 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> page = 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> totalNumberOfItems = 1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 13</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   snpId    pos snpIdUpper variantId geneSymbol  pValue geneSymbolUpper datasetId</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>           <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> rs14… 1.53<span style="color: #949494;">e</span>8 RS1410858  chr1_153… LOR        5.47<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> LOR             gtex_v8  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 5 more variables: tissueSiteDetailId &lt;chr&gt;, ontologyId &lt;chr&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   chromosome &lt;chr&gt;, gencodeId &lt;chr&gt;, nes &lt;dbl&gt;</span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick, Benjamin Zuckerman, Abraham Olvera-Barrios, Chuin Ying Ung, Robert Luben.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

