<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get Clustered Median Junction Expression — get_clustered_median_junction_expression • gtexr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Clustered Median Junction Expression — get_clustered_median_junction_expression"><meta name="description" content="Find median junction expression data along with hierarchical
clusters.
Returns median junction read counts in tissues of a given gene from all known transcripts along with the hierarchical clustering results of tissues and genes, based on junction expression, in Newick format.
Results may be filtered by dataset, gene or tissue, but at least one gene must be provided.
The hierarchical clustering is performed by calculating Euclidean distances and using the average linkage method.
This endpoint is not paginated.


By default, this service queries the latest GTEx release.
GTEx Portal API documentation"><meta property="og:description" content="Find median junction expression data along with hierarchical
clusters.
Returns median junction read counts in tissues of a given gene from all known transcripts along with the hierarchical clustering results of tissues and genes, based on junction expression, in Newick format.
Results may be filtered by dataset, gene or tissue, but at least one gene must be provided.
The hierarchical clustering is performed by calculating Euclidean distances and using the average linkage method.
This endpoint is not paginated.


By default, this service queries the latest GTEx release.
GTEx Portal API documentation"><meta property="og:image" content="https://rmgpanw.github.io/gtexr/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gtexr.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/developer_guide.html">Developer guide</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/rmgpanw/gtexr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get Clustered Median Junction Expression</h1>
      <small class="dont-index">Source: <a href="https://github.com/rmgpanw/gtexr/blob/main/R/get_clustered_median_junction_expression.R" class="external-link"><code>R/get_clustered_median_junction_expression.R</code></a></small>
      <div class="d-none name"><code>get_clustered_median_junction_expression.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Find median junction expression data along with hierarchical
clusters.</p><ul><li><p>Returns median junction read counts in tissues of a given gene from all known transcripts along with the hierarchical clustering results of tissues and genes, based on junction expression, in Newick format.</p></li>
<li><p>Results may be filtered by dataset, gene or tissue, but at least one gene must be provided.</p></li>
<li><p>The hierarchical clustering is performed by calculating Euclidean distances and using the average linkage method.</p></li>
<li><p><strong>This endpoint is not paginated.</strong></p></li>
</ul><p>By default, this service queries the latest GTEx release.</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Expression-Data-Endpoints/operation/get_clustered_median_junction_expression_api_v2_expression_clusteredMedianJunctionExpression_get" class="external-link">GTEx Portal API documentation</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_clustered_median_junction_expression</span><span class="op">(</span></span>
<span>  <span class="va">gencodeIds</span>,</span>
<span>  datasetId <span class="op">=</span> <span class="st">"gtex_v8"</span>,</span>
<span>  tissueSiteDetailIds <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gencodeids">gencodeIds<a class="anchor" aria-label="anchor" href="#arg-gencodeids"></a></dt>
<dd><p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p></dd>


<dt id="arg-datasetid">datasetId<a class="anchor" aria-label="anchor" href="#arg-datasetid"></a></dt>
<dd><p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p></dd>


<dt id="arg-tissuesitedetailids">tissueSiteDetailIds<a class="anchor" aria-label="anchor" href="#arg-tissuesitedetailids"></a></dt>
<dd><p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code><a href="get_tissue_site_detail.html">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A tibble, with clustering data stored as an attribute, "clusters".</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Expression Data Endpoints:
<code><a href="get_clustered_median_exon_expression.html">get_clustered_median_exon_expression</a>()</code>,
<code><a href="get_clustered_median_gene_expression.html">get_clustered_median_gene_expression</a>()</code>,
<code><a href="get_clustered_median_transcript_expression.html">get_clustered_median_transcript_expression</a>()</code>,
<code><a href="get_expression_pca.html">get_expression_pca</a>()</code>,
<code><a href="get_gene_expression.html">get_gene_expression</a>()</code>,
<code><a href="get_median_exon_expression.html">get_median_exon_expression</a>()</code>,
<code><a href="get_median_gene_expression.html">get_median_gene_expression</a>()</code>,
<code><a href="get_median_junction_expression.html">get_median_junction_expression</a>()</code>,
<code><a href="get_median_transcript_expression.html">get_median_transcript_expression</a>()</code>,
<code><a href="get_single_nucleus_gex.html">get_single_nucleus_gex</a>()</code>,
<code><a href="get_single_nucleus_gex_summary.html">get_single_nucleus_gex_summary</a>()</code>,
<code><a href="get_top_expressed_genes.html">get_top_expressed_genes</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu">get_clustered_median_junction_expression</span><span class="op">(</span>gencodeIds <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000203782.5"</span>,</span></span>
<span class="r-in"><span>                                                    <span class="st">"ENSG00000132693.12"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Retrieve clustering data with `attr(&lt;df&gt;, 'clusters')`</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 432 × 8</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    median junctionId           tissueSiteDetailId ontologyId datasetId gencodeId</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span>      0 chr1_159712581_1597… Adipose_Subcutane… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span>      0 chr1_159712795_1597… Adipose_Subcutane… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span>      0 chr1_159712795_1597… Adipose_Subcutane… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span>      0 chr1_159713604_1597… Adipose_Subcutane… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span>      0 chr1_159713641_1597… Adipose_Subcutane… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span>      0 chr1_159713973_1597… Adipose_Subcutane… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span>      0 chr1_159714139_1597… Adipose_Subcutane… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span>      0 chr1_159712581_1597… Adipose_Visceral_… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span>      0 chr1_159712795_1597… Adipose_Visceral_… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span>      0 chr1_159712795_1597… Adipose_Visceral_… UBERON:00… gtex_v8   ENSG0000…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 422 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 2 more variables: geneSymbol &lt;chr&gt;, unit &lt;chr&gt;</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># clustering data is stored as an attribute "clusters"</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">get_clustered_median_junction_expression</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000203782.5"</span>,</span></span>
<span class="r-in"><span>                                                 <span class="st">"ENSG00000132693.12"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Retrieve clustering data with `attr(&lt;df&gt;, 'clusters')`</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">result</span>, <span class="st">"clusters"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $junction</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "((((((chr1_159713641_159714006:0.03,chr1_159713604_159714002:0.03):0.67,(chr1_159712795_159714006:0.01,chr1_159712581_159713502:0.01):0.69):0.56,chr1_159712795_159713502:1.26):0.90,chr1_159713973_159714498:2.16):0.54,chr1_159714139_159714424:2.70):2.61,chr1_153259734_153260926:5.31);"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $tissue</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "(((((((((((((((((Adipose_Visceral_Omentum:0.00,Adipose_Subcutaneous:0.00):0.00,Artery_Aorta:0.00):0.00,Artery_Coronary:0.00):0.00,Artery_Tibial:0.00):0.00,Brain_Cortex:0.00):0.00,Breast_Mammary_Tissue:0.00):0.00,Colon_Sigmoid:0.00):0.00,Heart_Atrial_Appendage:0.00):0.00,Muscle_Skeletal:0.00):0.00,Nerve_Tibial:0.00):0.00,Vagina:0.00):0.08,Kidney_Medulla:0.08):0.03,(((((((((((((((((((((Brain_Amygdala:0.00,Bladder:0.00):0.00,Brain_Anterior_cingulate_cortex_BA24:0.00):0.00,Brain_Caudate_basal_ganglia:0.00):0.00,Brain_Cerebellar_Hemisphere:0.00):0.00,Brain_Cerebellum:0.00):0.00,Brain_Frontal_Cortex_BA9:0.00):0.00,Brain_Hippocampus:0.00):0.00,Brain_Hypothalamus:0.00):0.00,Brain_Nucleus_accumbens_basal_ganglia:0.00):0.00,Brain_Putamen_basal_ganglia:0.00):0.00,Brain_Spinal_cord_cervical_c-1:0.00):0.00,Brain_Substantia_nigra:0.00):0.00,Cells_EBV-transformed_lymphocytes:0.00):0.00,Colon_Transverse:0.00):0.00,Esophagus_Gastroesophageal_Junction:0.00):0.00,Esophagus_Muscularis:0.00):0.00,Heart_Left_Ventricle:0.00):0.00,Kidney_Cortex:0.00):0.00,Pituitary:0.00):0.00,Thyroid:0.00):0.00,Whole_Blood:0.00):0.11):0.03,((((((((((((Cells_Cultured_fibroblasts:0.00,Adrenal_Gland:0.00):0.00,Esophagus_Mucosa:0.00):0.00,Fallopian_Tube:0.00):0.00,Lung:0.00):0.00,Minor_Salivary_Gland:0.00):0.00,Ovary:0.00):0.00,Prostate:0.00):0.00,Small_Intestine_Terminal_Ileum:0.00):0.00,Spleen:0.00):0.00,Stomach:0.00):0.00,Uterus:0.00):0.06,((Cervix_Endocervix:0.00,Cervix_Ectocervix:0.00):0.00,Testis:0.00):0.06):0.09):0.19,Pancreas:0.33):0.27,(Skin_Sun_Exposed_Lower_leg:0.02,Skin_Not_Sun_Exposed_Suprapubic:0.02):0.59):0.68,Liver:1.29);"</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># process clustering data with the ape package</span></span></span>
<span class="r-in"><span><span class="co"># install.packages("ape")</span></span></span>
<span class="r-in"><span><span class="co"># phylo_tree &lt;- ape::read.tree(text = attr(result, "clusters")$tissue)</span></span></span>
<span class="r-in"><span><span class="co"># plot(phylo_tree)</span></span></span>
<span class="r-in"><span><span class="co"># print(phylo_tree)</span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick, Benjamin Zuckerman, Abraham Olvera-Barrios, Chuin Ying Ung, Robert Luben.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

