<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Find multi-tissue eQTL Metasoft results.
This service returns multi-tissue eQTL Metasoft results for a given gene and variant in a specified dataset.
A Versioned GENCODE ID must be provided.
For each tissue, the results include: m-value (mValue), normalized effect size (nes), p-value (pValue), and standard error (se).
The m-value is the posterior probability that an eQTL effect exists in each tissue tested in the cross-tissue meta-analysis (Han and Eskin, PLoS Genetics 8(3): e1002555, 2012).
The normalized effect size is the slope of the linear regression of normalized expression data versus the three genotype categories using single-tissue eQTL analysis, representing eQTL effect size.
The p-value is from a t-test that compares observed NES from single-tissue eQTL analysis to a null NES of 0.


By default, the service queries the latest GTEx release. The retrieved data is split into pages with items_per_page entries per page
GTEx Portal API documentation"><title>Get Multi Tissue Eqtls — get_multi_tissue_eqtls • gtexr</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Multi Tissue Eqtls — get_multi_tissue_eqtls"><meta property="og:description" content="Find multi-tissue eQTL Metasoft results.
This service returns multi-tissue eQTL Metasoft results for a given gene and variant in a specified dataset.
A Versioned GENCODE ID must be provided.
For each tissue, the results include: m-value (mValue), normalized effect size (nes), p-value (pValue), and standard error (se).
The m-value is the posterior probability that an eQTL effect exists in each tissue tested in the cross-tissue meta-analysis (Han and Eskin, PLoS Genetics 8(3): e1002555, 2012).
The normalized effect size is the slope of the linear regression of normalized expression data versus the three genotype categories using single-tissue eQTL analysis, representing eQTL effect size.
The p-value is from a t-test that compares observed NES from single-tissue eQTL analysis to a null NES of 0.


By default, the service queries the latest GTEx release. The retrieved data is split into pages with items_per_page entries per page
GTEx Portal API documentation"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/rmgpanw/gtexr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Get Multi Tissue Eqtls</h1>
      <small class="dont-index">Source: <a href="https://github.com/rmgpanw/gtexr/blob/HEAD/R/get_multi_tissue_eqtls.R" class="external-link"><code>R/get_multi_tissue_eqtls.R</code></a></small>
      <div class="d-none name"><code>get_multi_tissue_eqtls.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Find multi-tissue eQTL Metasoft results.</p><ul><li><p>This service returns multi-tissue eQTL Metasoft results for a given gene and variant in a specified dataset.</p></li>
<li><p>A Versioned GENCODE ID must be provided.</p></li>
<li><p>For each tissue, the results include: m-value (mValue), normalized effect size (nes), p-value (pValue), and standard error (se).</p></li>
<li><p>The m-value is the posterior probability that an eQTL effect exists in each tissue tested in the cross-tissue meta-analysis (Han and Eskin, PLoS Genetics 8(3): e1002555, 2012).</p></li>
<li><p>The normalized effect size is the slope of the linear regression of normalized expression data versus the three genotype categories using single-tissue eQTL analysis, representing eQTL effect size.</p></li>
<li><p>The p-value is from a t-test that compares observed NES from single-tissue eQTL analysis to a null NES of 0.</p></li>
</ul><p>By default, the service queries the latest GTEx release. The retrieved data is split into pages with items_per_page entries per page</p>
<p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_multi_tissue_eqtls_api_v2_association_metasoft_get" class="external-link">GTEx Portal API documentation</a></p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_multi_tissue_eqtls</span><span class="op">(</span></span>
<span>  <span class="va">gencodeIds</span>,</span>
<span>  variantId <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  datasetId <span class="op">=</span> <span class="st">"gtex_v8"</span>,</span>
<span>  page <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  itemsPerPage <span class="op">=</span> <span class="fl">250</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>gencodeIds</dt>
<dd><p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p></dd>


<dt>variantId</dt>
<dd><p>String. A gtex variant ID.</p></dd>


<dt>datasetId</dt>
<dd><p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".
Default = "gtex_v8".</p></dd>


<dt>page</dt>
<dd><p>Integer (default = 0).</p></dd>


<dt>itemsPerPage</dt>
<dd><p>Integer (default = 250).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A tibble.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Static Association Endpoints: 
<code><a href="get_eqtl_genes.html">get_eqtl_genes</a>()</code>,
<code><a href="get_fine_mapping.html">get_fine_mapping</a>()</code>,
<code><a href="get_independent_eqtl.html">get_independent_eqtl</a>()</code>,
<code><a href="get_significant_single_tissue_eqtls_by_location.html">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="get_significant_single_tissue_eqtls.html">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="get_significant_single_tissue_Ieqtls.html">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="get_significant_single_tissue_isqtls.html">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="get_significant_single_tissue_sqtls.html">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="get_sqtl_genes.html">get_sqtl_genes</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># search by gene</span></span></span>
<span class="r-in"><span><span class="fu">get_multi_tissue_eqtls</span><span class="op">(</span>gencodeId <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENSG00000132693.12"</span>,</span></span>
<span class="r-in"><span>                                     <span class="st">"ENSG00000203782.5"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> numberOfPages = 1</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> page = 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> totalNumberOfItems = 132</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 6,468 × 5</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    gencodeId         datasetId   metaP variantId              tissues         </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;named list&gt;</span>    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> ENSG00000203782.5 gtex_v8   0.001<span style="text-decoration: underline;">77</span> chr1_152346526_G_T_b38 <span style="color: #949494;">&lt;named list [4]&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 6,458 more rows</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

