<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Get Eqtl Genes — get_eqtl_genes • gtexr</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Get Eqtl Genes — get_eqtl_genes"><meta name="description" content="Retrieve eGenes (eQTL Genes).
This service returns eGenes (eQTL Genes) from the specified dataset.
eGenes are genes that have at least one significant cis-eQTL acting upon them.
Results may be filtered by tissue. By default, the service queries the latest GTEx release.


For each eGene, the results include the allelic fold change
(log2AllelicFoldChange), p-value (pValue), p-value threshold
(pValueThreshold), empirical p-value (empiricalPValue), and q-value
(qValue).
The log2AllelicFoldChange is the allelic fold change (in log2 scale) of the most significant eQTL.
The pValue is the nominal p-value of the most significant eQTL.
The pValueThreshold is the p-value threshold used to determine whether a cis-eQTL for this gene is significant. For more details see https://gtexportal.org/home/documentationPage#staticTextAnalysisMethods.
The empiricalPValue is the beta distribution-adjusted empirical p-value from FastQTL.
The qValues were calculated based on the empirical p-values. A false discovery rate (FDR) threshold of &amp;lt;= 0.05 was applied to identify genes with a significant eQTL.


GTEx Portal API documentation."><meta property="og:description" content="Retrieve eGenes (eQTL Genes).
This service returns eGenes (eQTL Genes) from the specified dataset.
eGenes are genes that have at least one significant cis-eQTL acting upon them.
Results may be filtered by tissue. By default, the service queries the latest GTEx release.


For each eGene, the results include the allelic fold change
(log2AllelicFoldChange), p-value (pValue), p-value threshold
(pValueThreshold), empirical p-value (empiricalPValue), and q-value
(qValue).
The log2AllelicFoldChange is the allelic fold change (in log2 scale) of the most significant eQTL.
The pValue is the nominal p-value of the most significant eQTL.
The pValueThreshold is the p-value threshold used to determine whether a cis-eQTL for this gene is significant. For more details see https://gtexportal.org/home/documentationPage#staticTextAnalysisMethods.
The empiricalPValue is the beta distribution-adjusted empirical p-value from FastQTL.
The qValues were calculated based on the empirical p-values. A false discovery rate (FDR) threshold of &amp;lt;= 0.05 was applied to identify genes with a significant eQTL.


GTEx Portal API documentation."><meta property="og:image" content="https://rmgpanw.github.io/gtexr/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/gtexr.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/developer_guide.html">Developer guide</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/rmgpanw/gtexr/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Get Eqtl Genes</h1>
      <small class="dont-index">Source: <a href="https://github.com/rmgpanw/gtexr/blob/v0.1.0.9000/R/get_eqtl_genes.R" class="external-link"><code>R/get_eqtl_genes.R</code></a></small>
      <div class="d-none name"><code>get_eqtl_genes.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Retrieve eGenes (eQTL Genes).</p><ul><li><p>This service returns eGenes (eQTL Genes) from the specified dataset.</p></li>
<li><p>eGenes are genes that have at least one significant cis-eQTL acting upon them.</p></li>
<li><p>Results may be filtered by tissue. By default, the service queries the latest GTEx release.</p></li>
</ul><p>For each eGene, the results include the allelic fold change
(log2AllelicFoldChange), p-value (pValue), p-value threshold
(pValueThreshold), empirical p-value (empiricalPValue), and q-value
(qValue).</p><ul><li><p>The log2AllelicFoldChange is the allelic fold change (in log2 scale) of the most significant eQTL.</p></li>
<li><p>The pValue is the nominal p-value of the most significant eQTL.</p></li>
<li><p>The pValueThreshold is the p-value threshold used to determine whether a cis-eQTL for this gene is significant. For more details see https://gtexportal.org/home/documentationPage#staticTextAnalysisMethods.</p></li>
<li><p>The empiricalPValue is the beta distribution-adjusted empirical p-value from FastQTL.</p></li>
<li><p>The qValues were calculated based on the empirical p-values. A false discovery rate (FDR) threshold of &lt;= 0.05 was applied to identify genes with a significant eQTL.</p></li>
</ul><p><a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_eqtl_genes_api_v2_association_egene_get" class="external-link">GTEx Portal API documentation</a>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">get_eqtl_genes</span><span class="op">(</span></span>
<span>  <span class="va">tissueSiteDetailIds</span>,</span>
<span>  datasetId <span class="op">=</span> <span class="st">"gtex_v8"</span>,</span>
<span>  page <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  itemsPerPage <span class="op">=</span> <span class="fl">250</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-tissuesitedetailids">tissueSiteDetailIds<a class="anchor" aria-label="anchor" href="#arg-tissuesitedetailids"></a></dt>
<dd><p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code><a href="get_tissue_site_detail.html">get_tissue_site_detail()</a></code> to see valid values) or Ontology IDs.</p></dd>


<dt id="arg-datasetid">datasetId<a class="anchor" aria-label="anchor" href="#arg-datasetid"></a></dt>
<dd><p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p></dd>


<dt id="arg-page">page<a class="anchor" aria-label="anchor" href="#arg-page"></a></dt>
<dd><p>Integer (default = 0).</p></dd>


<dt id="arg-itemsperpage">itemsPerPage<a class="anchor" aria-label="anchor" href="#arg-itemsperpage"></a></dt>
<dd><p>Integer (default = 250).</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A tibble.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other Static Association Endpoints:
<code><a href="get_fine_mapping.html">get_fine_mapping</a>()</code>,
<code><a href="get_independent_eqtl.html">get_independent_eqtl</a>()</code>,
<code><a href="get_multi_tissue_eqtls.html">get_multi_tissue_eqtls</a>()</code>,
<code><a href="get_significant_single_tissue_eqtls.html">get_significant_single_tissue_eqtls</a>()</code>,
<code><a href="get_significant_single_tissue_eqtls_by_location.html">get_significant_single_tissue_eqtls_by_location</a>()</code>,
<code><a href="get_significant_single_tissue_ieqtls.html">get_significant_single_tissue_ieqtls</a>()</code>,
<code><a href="get_significant_single_tissue_isqtls.html">get_significant_single_tissue_isqtls</a>()</code>,
<code><a href="get_significant_single_tissue_sqtls.html">get_significant_single_tissue_sqtls</a>()</code>,
<code><a href="get_sqtl_genes.html">get_sqtl_genes</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu">get_eqtl_genes</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Whole_Blood"</span>, <span class="st">"Artery_Aorta"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span class="warning">Warning: </span><span style="color: #BBBB00;">!</span> Total number of items (24853) exceeds maximum page size (250).</span>
<span class="r-wrn co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> Try increasing `itemsPerPage`.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> numberOfPages = 100</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> page = 0</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">•</span> totalNumberOfItems = 24853</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 250 × 10</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    tissueSiteDetailId ontologyId  datasetId empiricalPValue gencodeId geneSymbol</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> Whole_Blood        UBERON:001… gtex_v8          1.05<span style="color: #949494;">e</span><span style="color: #BB0000;">- 9</span> ENSG0000… WASH7P    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> Whole_Blood        UBERON:001… gtex_v8          1.06<span style="color: #949494;">e</span><span style="color: #BB0000;">-25</span> ENSG0000… RP11-34P1…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> Whole_Blood        UBERON:001… gtex_v8          6.31<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> ENSG0000… CICP27    </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> Whole_Blood        UBERON:001… gtex_v8          8.71<span style="color: #949494;">e</span><span style="color: #BB0000;">- 9</span> ENSG0000… RP11-34P1…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> Whole_Blood        UBERON:001… gtex_v8          6.01<span style="color: #949494;">e</span><span style="color: #BB0000;">-20</span> ENSG0000… RP11-34P1…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> Whole_Blood        UBERON:001… gtex_v8          6.96<span style="color: #949494;">e</span><span style="color: #BB0000;">- 9</span> ENSG0000… RP11-34P1…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> Whole_Blood        UBERON:001… gtex_v8          3.10<span style="color: #949494;">e</span><span style="color: #BB0000;">- 4</span> ENSG0000… RP11-34P1…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> Whole_Blood        UBERON:001… gtex_v8          1.92<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span> ENSG0000… ABC7-4304…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> Whole_Blood        UBERON:001… gtex_v8          1.58<span style="color: #949494;">e</span><span style="color: #BB0000;">- 3</span> ENSG0000… RP11-34P1…</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> Whole_Blood        UBERON:001… gtex_v8          7.82<span style="color: #949494;">e</span><span style="color: #BB0000;">- 2</span> ENSG0000… AP006222.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 240 more rows</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 4 more variables: log2AllelicFoldChange &lt;dbl&gt;, pValue &lt;dbl&gt;,</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;">#   pValueThreshold &lt;dbl&gt;, qValue &lt;dbl&gt;</span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick, Benjamin Zuckerman, Abraham Olvera-Barrios, Chuin Ying Ung, Robert Luben.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

