<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Internal function that documents all arguments for exported gtexr functions
that wrap GTEx Portal API endpoints (using roxygen @inheritParams tag)."><title>gtexr arguments — gtexr_arguments • gtexr</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="gtexr arguments — gtexr_arguments"><meta property="og:description" content="Internal function that documents all arguments for exported gtexr functions
that wrap GTEx Portal API endpoints (using roxygen @inheritParams tag)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/rmgpanw/gtexr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>gtexr arguments</h1>
      <small class="dont-index">Source: <a href="https://github.com/rmgpanw/gtexr/blob/HEAD/R/gtexr_arguments.R" class="external-link"><code>R/gtexr_arguments.R</code></a></small>
      <div class="d-none name"><code>gtexr_arguments.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Internal function that documents all arguments for exported gtexr functions
that wrap GTEx Portal API endpoints (using roxygen <code>@inheritParams</code> tag).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">gtexr_arguments</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>ageBrackets</dt>
<dd><p>The age bracket(s) of the donors of interest. Options:
"20-29", "30-39", "40-49", "50-59", "60-69", "70-79".</p></dd>


<dt>aliquotIds</dt>
<dd><p>Character vector.</p></dd>


<dt>autolysisScores</dt>
<dd><p>Character vector. Options: "None", "Mild", "Moderate",
"Severe".</p></dd>


<dt>attributeSubset</dt>
<dd><p>String. Examples include but are not limited to "sex", "ageBracket"</p></dd>


<dt>bp_window</dt>
<dd><p>Integer.</p></dd>


<dt>cellType</dt>
<dd><p>String. "Adipocytes", "Epithelial_cells", "Hepatocytes",
"Keratinocytes", "Myocytes", "Neurons", "Neutrophils".</p></dd>


<dt>chromosome</dt>
<dd><p>String. One of "chr1", "chr2", "chr3", "chr4", "chr5",
"chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13",
"chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21",
"chr22", "chrM", "chrX", "chrY".</p></dd>


<dt>datasetId</dt>
<dd><p>String. Unique identifier of a dataset. Usually includes a
data source and data release. Options: "gtex_v8", "gtex_snrnaseq_pilot".</p></dd>


<dt>dataTypes</dt>
<dd><p>Character vector. Options: "RNASEQ", "WGS", "WES", "OMNI",
"EXCLUDE".</p></dd>


<dt>end</dt>
<dd><p>Integer.</p></dd>


<dt>excludeDataArray</dt>
<dd><p>String. Options are TRUE or FALSE</p></dd>


<dt>filterMtGene</dt>
<dd><p>Logical. Exclude mitochondrial genes.</p></dd>


<dt>gencodeId</dt>
<dd><p>String. A Versioned GENCODE ID of a gene, e.g.
"ENSG00000065613.9".</p></dd>


<dt>gencodeIds</dt>
<dd><p>A character vector of Versioned GENCODE IDs, e.g.
c("ENSG00000132693.12", "ENSG00000203782.5").</p></dd>


<dt>gencodeVersion</dt>
<dd><p>String (default = "v26"). GENCODE annotation release.
Either "v26" or "v19".</p></dd>


<dt>geneId</dt>
<dd><p>String. A gene symbol, a gencode ID, or an Ensemble ID.</p></dd>


<dt>geneIds</dt>
<dd><p>A character vector of gene symbols, versioned gencodeIds, or
unversioned gencodeIds.</p></dd>


<dt>genomeBuild</dt>
<dd><p>String. Options: "GRCh38/hg38", "GRCh37/hg19". Default =
"GRCh38/hg38".</p></dd>


<dt>hardyScales</dt>
<dd><p>Character vector. A list of Hardy Scale(s) of interest.
Options: "Ventilator case", "Fast death - violent", "Fast death - natural
causes", "Intermediate death", "Slow death".</p></dd>


<dt>itemsPerPage</dt>
<dd><p>Integer (default = 250).</p></dd>


<dt>ischemicTime</dt>
<dd><p>Integer.</p></dd>


<dt>ischemicTimeGroups</dt>
<dd><p>Character vector. Options: "&lt;= 0", "1 - 300", "301</p><ul><li><p>600", "601 - 900", "901 - 1200", "1201 - 1500", "&gt; 1500".</p></li>
</ul></dd>


<dt>organizationName</dt>
<dd><p>String. Options: "GTEx Consortium" "Kid's First".</p></dd>


<dt>page</dt>
<dd><p>Integer (default = 0).</p></dd>


<dt>pathCategory</dt>
<dd><p>Character vector. Options: "adenoma", "amylacea",
"atelectasis", "atherosclerosis", "atherosis", "atrophy", "calcification",
"cirrhosis", "clean_specimens", "congestion", "corpora_albicantia", "cyst",
"desquamation", "diabetic", "dysplasia", "edema", "emphysema",
"esophagitis", "fibrosis", "gastritis", "glomerulosclerosis", "goiter",
"gynecomastoid", "hashimoto", "heart_failure_cells", "hemorrhage",
"hepatitis", "hyalinization", "hypereosinophilia", "hyperplasia",
"hypertrophy", "hypoxic", "infarction", "inflammation", "ischemic_changes",
"macrophages", "mastopathy", "metaplasia", "monckeberg", "necrosis",
"nephritis", "nephrosclerosis", "no_abnormalities", "nodularity",
"pancreatitis", "pigment", "pneumonia", "post_menopausal", "prostatitis",
"saponification", "scarring", "sclerotic", "solar_elastosis",
"spermatogenesis", "steatosis", "sweat_glands", "tma".</p></dd>


<dt>pos</dt>
<dd><p>Integer, vector.</p></dd>


<dt>project_id</dt>
<dd><p>String. Options: "gtex", "adult-gtex", "egtex".</p></dd>


<dt>rin</dt>
<dd><p>Integer vector.</p></dd>


<dt>sampleId</dt>
<dd><p>String. <code>^GTEX-[A-Z0-9]{5}-[0-9]{4}-SM-[A-Z0-9]{5}$</code></p></dd>


<dt>sampleIds</dt>
<dd><p>Character vector. GTEx sample ID.</p></dd>


<dt>sex</dt>
<dd><p>String. Options: "male", "female".</p></dd>


<dt>snpId</dt>
<dd><p>String</p></dd>


<dt>sortBy</dt>
<dd><p>String. Options: "sampleId", "ischemicTime", "aliquotId",
"tissueSampleId", "hardyScale", "pathologyNotes", "ageBracket",
"tissueSiteDetailId", "sex".</p></dd>


<dt>sortDirection</dt>
<dd><p>String. Options: "asc", "desc". Default = "asc".</p></dd>


<dt>start</dt>
<dd><p>Integer.</p></dd>


<dt>subjectIds</dt>
<dd><p>Character vector. GTEx subject ID.</p></dd>


<dt>tissueSampleIds</dt>
<dd><p>Array of strings. A list of Tissue Sample ID(s).</p></dd>


<dt>tissueSiteDetailId</dt>
<dd><p>String. The ID of the tissue of interest. Can be a
GTEx specific ID (e.g. "Whole_Blood"; use <code><a href="available_tissueSiteDetailIds.html">available_tissueSiteDetailIds()</a></code>
to see valid values) or an Ontology ID.</p></dd>


<dt>tissueSiteDetailIds</dt>
<dd><p>Character vector of IDs for tissues of interest.
Can be GTEx specific IDs (e.g. "Whole_Blood"; use
<code><a href="available_tissueSiteDetailIds.html">available_tissueSiteDetailIds()</a></code> to see valid values) or Ontology IDs.</p></dd>


<dt>uberonIds</dt>
<dd><p>Character vector. A list of Uberon ID(s) of interest.
Options: "UBERON:EFO_0000572", "UBERON:EFO_0002009", "UBERON:0000007",
"UBERON:0000458", "UBERON:0000473", "UBERON:0000945", "UBERON:0000992",
"UBERON:0000995", "UBERON:0000996", "UBERON:0001114", "UBERON:0001150",
"UBERON:0001157", "UBERON:0001159", "UBERON:0001211", "UBERON:0001225",
"UBERON:0001255", "UBERON:0001293", "UBERON:0001323", "UBERON:0001496",
"UBERON:0001621", "UBERON:0001870", "UBERON:0001873", "UBERON:0001874",
"UBERON:0001876", "UBERON:0001882", "UBERON:0001898", "UBERON:0001954",
"UBERON:0002037", "UBERON:0002038", "UBERON:0002046", "UBERON:0002106",
"UBERON:0002190", "UBERON:0002367", "UBERON:0002369", "UBERON:0003889",
"UBERON:0004264", "UBERON:0004550", "UBERON:0004648", "UBERON:0006330",
"UBERON:0006469", "UBERON:0006566", "UBERON:0006631", "UBERON:0006920",
"UBERON:0007610", "UBERON:0008367", "UBERON:0008952", "UBERON:0009834",
"UBERON:0009835", "UBERON:0010414", "UBERON:0011907", "UBERON:0012249",
"UBERON:0013756", "UBERON:0036149".</p></dd>


<dt>variantId</dt>
<dd><p>String. A gtex variant ID.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Returns <code>NULL</code> invisibly. Used for documentation only.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

