<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gtexr">
<title>Introduction to gtexr • gtexr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to gtexr">
<meta property="og:description" content="gtexr">
<meta property="og:image" content="https://rmgpanw.github.io/gtexr/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gtexr</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/gtexr.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/developer_guide.html">Developer guide</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/rmgpanw/gtexr/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Introduction to gtexr</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/rmgpanw/gtexr/blob/HEAD/vignettes/gtexr.Rmd" class="external-link"><code>vignettes/gtexr.Rmd</code></a></small>
      <div class="d-none name"><code>gtexr.Rmd</code></div>
    </div>

    
    
<p>The <a href="https://gtexportal.org/home/apiPage" class="external-link">GTEx Portal API
V2</a> enables programmatic access to data available from the
Genotype-Tissue Expression Portal. The gtexr package wraps this API,
providing R functions that correspond to each <a href="https://gtexportal.org/api/v2/redoc" class="external-link">API endpoint</a>:</p>
<ul>
<li>R function names mirror those of their corresponding endpoint,
converted to lower case with spaces replaced with underscores e.g. the R
function for <a href="https://gtexportal.org/api/v2/redoc#tag/GTEx-Portal-API-Info/operation/get_service_info_api_v2__get" class="external-link">“Get
Service Info”</a> is <code><a href="../reference/get_service_info.html">get_service_info()</a></code>.</li>
<li>Query parameters are similarly mirrored by function arguments
e.g. the arguments for <code><a href="../reference/get_maintenance_message.html">get_maintenance_message()</a></code>
(corresponding to the endpoint <a href="https://gtexportal.org/api/v2/redoc#tag/Admin-Endpoints/operation/get_maintenance_message_api_v2_admin_maintenanceMessage_get" class="external-link">“Get
Maintenance Message”</a>) are <code>page</code> and
<code>itemsPerPage</code>. For query parameters that accept an array of
values however, the corresponding function argument is pluralised to
indicate this e.g. for endpoint <a href="https://gtexportal.org/api/v2/redoc#tag/Static-Association-Endpoints/operation/get_eqtl_genes_api_v2_association_egene_get" class="external-link">“Get
Eqtl Genes”</a> the query parameter ‘tissueSiteDetailId’ is pluralised
to argument name <code>tissueSiteDetailIds</code> in
<code><a href="../reference/get_eqtl_genes.html">get_eqtl_genes()</a></code>.</li>
<li>Default values for arguments mirror those for the API</li>
<li>The documentation for each function includes at least one working
example e.g. <code><a href="../reference/get_eqtl_genes.html">?get_eqtl_genes</a></code> provides example valid values
for the required argument <code>tissueSiteDetailIds</code>.</li>
<li>All functions return a <code><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble::tibble</a></code>.</li>
</ul>
<div class="section level2">
<h2 id="shiny-app">Shiny app<a class="anchor" aria-label="anchor" href="#shiny-app"></a>
</h2>
<p>Users can try out all functions interatively with the ⭐<a href="https://7hocgq-rmgpanw.shinyapps.io/gtexr/" class="external-link">gtexr shiny app</a>⭐,
which pre-populates query parameters with those for the first working
example from each function’s documentation.</p>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>The rest of this vignette outlines some example applications of
gtexr.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://rmgpanw.github.io/gtexr/">gtexr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://purrr.tidyverse.org/" class="external-link">purrr</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="get-build-37-coordinates-for-a-variant">Get build 37 coordinates for a variant<a class="anchor" aria-label="anchor" href="#get-build-37-coordinates-for-a-variant"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_variant.html">get_variant</a></span><span class="op">(</span>snpId <span class="op">=</span> <span class="st">"rs1410858"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span></span>
<span>    col <span class="op">=</span> <span class="va">b37VariantId</span>,</span>
<span>    into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      <span class="st">"chromosome"</span>,</span>
<span>      <span class="st">"position"</span>,</span>
<span>      <span class="st">"reference_allele"</span>,</span>
<span>      <span class="st">"alternative_allele"</span>,</span>
<span>      <span class="st">"genome_build"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    sep <span class="op">=</span> <span class="st">"_"</span>,</span>
<span>    remove <span class="op">=</span> <span class="cn">FALSE</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">snpId</span><span class="op">:</span><span class="va">genome_build</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> numberOfPages = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> page = 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> totalNumberOfItems = 1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 7</span></span></span>
<span><span class="co">#&gt;   snpId     b37VariantId chromosome position reference_allele alternative_allele</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> rs1410858 1_153182116… 1          1531821… C                A                 </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: genome_build &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="convert-gene-symbol-to-versioned-gencode-id">Convert gene symbol to versioned GENCODE ID<a class="anchor" aria-label="anchor" href="#convert-gene-symbol-to-versioned-gencode-id"></a>
</h3>
<p>Use <code>get_gene()</code> or <code><a href="../reference/get_genes.html">get_genes()</a></code></p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_genes.html">get_genes</a></span><span class="op">(</span><span class="st">"CRP"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">geneSymbol</span>, <span class="va">gencodeId</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> numberOfPages = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> page = 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> totalNumberOfItems = 1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 2</span></span></span>
<span><span class="co">#&gt;   geneSymbol gencodeId         </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CRP        ENSG00000132693.12</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="convert-rsid-to-gtex-variant-id">Convert rsID to GTEx variant ID<a class="anchor" aria-label="anchor" href="#convert-rsid-to-gtex-variant-id"></a>
</h3>
<p>Use <code><a href="../reference/get_variant.html">get_variant()</a></code></p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/get_variant.html">get_variant</a></span><span class="op">(</span>snpId <span class="op">=</span> <span class="st">"rs1410858"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">snpId</span>, <span class="va">variantId</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> numberOfPages = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> page = 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> totalNumberOfItems = 1</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 2</span></span></span>
<span><span class="co">#&gt;   snpId     variantId             </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> rs1410858 chr1_153209640_C_A_b38</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="for-a-gene-of-interest-which-tissues-have-significant-cis-eqtls">For a gene of interest, which tissues have significant
<em>cis</em>-eQTLs?<a class="anchor" aria-label="anchor" href="#for-a-gene-of-interest-which-tissues-have-significant-cis-eqtls"></a>
</h3>
<p>Use <code><a href="../reference/get_significant_single_tissue_eqtls.html">get_significant_single_tissue_eqtls()</a></code> (note this
requires versioned GENCODE IDs)</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_symbol_of_interest</span> <span class="op">&lt;-</span> <span class="st">"CRP"</span></span>
<span></span>
<span><span class="va">gene_gencodeId_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_genes.html">get_genes</a></span><span class="op">(</span><span class="va">gene_symbol_of_interest</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">gencodeId</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> numberOfPages = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> page = 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> totalNumberOfItems = 1</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">gene_gencodeId_of_interest</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/get_significant_single_tissue_eqtls.html">get_significant_single_tissue_eqtls</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/distinct.html" class="external-link">distinct</a></span><span class="op">(</span><span class="va">geneSymbol</span>, <span class="va">gencodeId</span>, <span class="va">tissueSiteDetailId</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Paging info</span> <span style="color: #00BBBB;">─────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> numberOfPages = 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> page = 0</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> maxItemsPerPage = 250</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">•</span> totalNumberOfItems = 93</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 3</span></span></span>
<span><span class="co">#&gt;   geneSymbol gencodeId          tissueSiteDetailId                 </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CRP        ENSG00000132693.12 Thyroid                            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> CRP        ENSG00000132693.12 Esophagus_Gastroesophageal_Junction</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> CRP        ENSG00000132693.12 Muscle_Skeletal</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="get-data-for-non-eqtl-variants">Get data for non-eQTL variants<a class="anchor" aria-label="anchor" href="#get-data-for-non-eqtl-variants"></a>
</h3>
<p>Some analyses (e.g. Mendelian randomisation) require data for
variants which may or may not be significant eQTLs. Use
<code><a href="../reference/calculate_expression_quantitative_trait_loci.html">calculate_expression_quantitative_trait_loci()</a></code> with
<code><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">purrr::map()</a></code> to retrieve data for multiple variants</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">variants_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"rs12119111"</span>, <span class="st">"rs6605071"</span>, <span class="st">"rs1053870"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">variants_of_interest</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://rlang.r-lib.org/reference/set_names.html" class="external-link">set_names</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span></span>
<span>    \<span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="../reference/calculate_expression_quantitative_trait_loci.html">calculate_expression_quantitative_trait_loci</a></span><span class="op">(</span></span>
<span>      tissueSiteDetailId <span class="op">=</span> <span class="st">"Liver"</span>,</span>
<span>      gencodeId <span class="op">=</span> <span class="st">"ENSG00000237973.1"</span>,</span>
<span>      variantId <span class="op">=</span> <span class="va">x</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span>.id <span class="op">=</span> <span class="st">"rsid"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  </span>
<span>  <span class="co"># optionally, reformat output - first extract genomic coordinates and alleles</span></span>
<span>  <span class="fu">tidyr</span><span class="fu">::</span><span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html" class="external-link">separate</a></span><span class="op">(</span></span>
<span>    col <span class="op">=</span> <span class="st">"variantId"</span>,</span>
<span>    into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>      <span class="st">"chromosome"</span>,</span>
<span>      <span class="st">"position"</span>,</span>
<span>      <span class="st">"reference_allele"</span>,</span>
<span>      <span class="st">"alternative_allele"</span>,</span>
<span>      <span class="st">"genome_build"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    sep <span class="op">=</span> <span class="st">"_"</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  </span>
<span>  <span class="co"># ...then ascertain alternative_allele frequency</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    alt_allele_count <span class="op">=</span> <span class="op">(</span><span class="fl">2</span> <span class="op">*</span> <span class="va">homoAltCount</span><span class="op">)</span> <span class="op">+</span> <span class="va">hetCount</span>,</span>
<span>    total_allele_count <span class="op">=</span> <span class="fl">2</span> <span class="op">*</span> <span class="op">(</span><span class="va">homoAltCount</span> <span class="op">+</span> <span class="va">hetCount</span> <span class="op">+</span>  <span class="va">homoRefCount</span><span class="op">)</span>,</span>
<span>    alternative_allele_frequency <span class="op">=</span> <span class="va">alt_allele_count</span> <span class="op">/</span> <span class="va">total_allele_count</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span></span>
<span>    <span class="va">rsid</span>,</span>
<span>    beta <span class="op">=</span> <span class="va">nes</span>,</span>
<span>    se <span class="op">=</span> <span class="va">error</span>,</span>
<span>    <span class="va">pValue</span>,</span>
<span>    minor_allele_frequency <span class="op">=</span> <span class="va">maf</span>,</span>
<span>    <span class="va">alternative_allele_frequency</span>,</span>
<span>    <span class="va">chromosome</span><span class="op">:</span><span class="va">genome_build</span>,</span>
<span>    <span class="va">tissueSiteDetailId</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 12</span></span></span>
<span><span class="co">#&gt;   rsid         beta     se  pValue minor_allele_frequency alternative_allele_f…¹</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> rs121191…  0.027<span style="text-decoration: underline;">0</span> 0.067<span style="text-decoration: underline;">0</span> 6.88<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span>                 0.365                   0.635</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> rs6605071 -<span style="color: #BB0000;">0.601</span>  0.166  3.88<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span>                 0.040<span style="text-decoration: underline;">9</span>                  0.959</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> rs1053870  0.024<span style="text-decoration: underline;">7</span> 0.073<span style="text-decoration: underline;">8</span> 7.38<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span>                 0.214                   0.214</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ abbreviated name: ¹​alternative_allele_frequency</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 6 more variables: chromosome &lt;chr&gt;, position &lt;chr&gt;, reference_allele &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   alternative_allele &lt;chr&gt;, genome_build &lt;chr&gt;, tissueSiteDetailId &lt;chr&gt;</span></span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Alasdair Warwick.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
